rotkehlchen/globaldb/upgrades/v2_v3.py

```diff
@@ -0,0 +1,361 @@
+import logging
+import sqlite3
+from collections import deque
+from typing import Deque, Dict, Tuple, Any, List
+import random
+
+# from eth_utils import to_checksum_address
+# import requests
+
+from rotkehlchen.constants.resolver import (
+    CHAINID,
+    EvmTokenKind,
+    VALID_EVM_CHAINS,
+    EVM_CHAINS_TO_DATABASE,
+    evm_address_to_identifier,
+    translate_old_format_to_new,
+)
+from rotkehlchen.globaldb.schema import (
+    DB_V3_CREATE_COMMON_ASSETS_DETAILS,
+    DB_V3_CREATE_ASSETS,
+    DB_V3_CREATE_EVM_TOKENS,
+    DB_V3_CREATE_ASSETS_EVM_TOKENS,
+    DB_V3_CREATE_MULTIASSETS,
+    DB_CREATE_ETHEREUM_TOKENS_LIST,
+    DB_CREATE_USER_OWNED_ASSETS,
+)
+
+log = logging.getLogger(__name__)
+
+
+COMMON_ASSETS_INSERT = (
+    """INSERT OR IGNORE INTO common_asset_details(
+        identifier, name, symbol, coingecko, cryptocompare, forked
+        ) VALUES(?, ?, ?, ?, ?, ?)"""
+)
+ASSETS_INSERT = (
+    """INSERT OR IGNORE INTO assets(
+            identifier, type, started, swapped_for, common_details_id
+        )VALUES(?, ?, ?, ?, ?);"""
+)
+EVM_TOKEN_INSERT = (
+    """INSERT OR IGNORE INTO evm_tokens(
+            identifier, token_type, chain, address, decimals, protocol
+        ) VALUES(?, ?, ?, ?, ?, ?)"""
+)
+UNDERLYING_TOKEN_INSERT = (
+    """INSERT OR IGNORE INTO underlying_tokens_list(identifier, weight, parent_token_entry)
+        VALUES (?, ?, ?)"""
+)
+OWNED_ASSETS_INSERT = 'INSERT OR IGNORE INTO user_owned_assets(asset_id) VALUES (?);'
+MIGRATION_CACHE: Deque[Any] = deque([])
+
+EVM_TUPLES_CREATION_TYPE = (
+    Tuple[
+        List[Tuple[Any, str, str, Any, Any, Any]],
+        List[Tuple[Any, str, Any, str, Any]],
+        List[Tuple[Any, Any, Any, Any, Any, None]],
+    ]
+)
+
+ASSET_CREATION_TYPE = (
+    Tuple[
+        List[Tuple[Any, str, Any, str, Any]],
+        List[Tuple[Any, Any, Any, Any, Any, None]],
+    ]
+)
+
+
+def coingecko_hack() -> Dict[str, str]:
+    """Avoid querying coingecko for now"""
+    # TODO ASSETS: Remove this hack
+    addr = '0x' + "".join(random.choices('0123456789ABCDEFabcdef', k=40))
+    return {k.value: addr for k in CHAINID}
+
+
+def query_coingecko_for_addresses(identifier: str) -> Dict[str, str]:  # noqa: E501 pylint: disable=unused-argument
```

Comment:
As discussed. This CAN NOT be part of the actual DB migration. This can only happen after the migration is complete, the user has logged in and a long-running task can executed.

Reply:
Yes I agree. Won't use it right here but needed a function to query ids. Will move this function to a different place since I don't use it